### PR TITLE
JENKINS-23485 - TokenMacro for getting a version value stored in environment variable

### PR DESCRIPTION
**Pull request by dawidmalina**

### CODE REVIEW

**Comment by dawidmalina** [comment]
@patbos this change is based on your pull request - JENKINS-23249 - implementation of POM_VERSION macro

**Comment by dawidmalina** [comment]
Remove maven-plugin dependency.

**Comment by dawidmalina** [comment]
Added more coverage

**Comment by patbos** [comment]
Thanks for the PR! Really wants this, but i feel like the maven-plugin is a better home for the functionality. There is a PR (https://github.com/jenkinsci/maven-plugin/pull/47) with this functionality for the maven-plugin.

**Comment by patbos** [comment]
But perhaps a way is to create a more general macro called like PARSE_VERSION that takes parses a environment variable name and optional parameter to strip snapshot.

**Comment by dawidmalina** [comment]
This is very good idea to propose this generic variable. I will provide this change next week (currently I'm out) or you can merge it as is and just rename this variable.

File: src/main/java/se/diabol/jenkins/pipeline/PipelineVersionContributor.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public Environment setUp(AbstractBuild build, Launcher launcher, BuildListener l
             InterruptedException {
         try {
 
-            String version = TokenMacro.expand(build, listener, getVersionTemplate());
+            String version = TokenMacro.expandAll(build, listener, getVersionTemplate());
```
Comment (dawidmalina) [inline src/main/java/se/diabol/jenkins/pipeline/PipelineVersionContributor.java]: This will allow to use all variables like this: `${ENV,var="NODE"}`

**Comment by warden** [comment]
+1 this will be useful and actually a better place than in maven-plugin since it can be used in freestyle projects.

**Comment by dawidmalina** [comment]
@patbos What do you think about this change?

**Comment by patbos** [comment]
Looks good! Im just missing help text for the macro. Create file:
src/main/resources/se/diabol/jenkins/pipeline/ParseVersionTokenMacro/help.jelly 
containing help text for the macro. See examples:
https://github.com/jenkinsci/git-plugin/blob/master/src/main/resources/hudson/plugins/git/GitBranchTokenMacro/help.jelly
https://github.com/jenkinsci/token-macro-plugin/blob/master/src/main/resources/org/jenkinsci/plugins/tokenmacro/impl/EnvironmentVariableMacro/help.jelly

**Comment by dawidmalina** [comment]
Done

**Comment by dawidmalina** [comment]
When you can merge this?

**Comment by patbos** [comment]
I looked at the issue and it was mentioning the name ENV_VERSION for the macro name. I think it is a better name than the PARSE_VERSION. What do you think?

**Comment by dawidmalina** [comment]
No problem I will change it tommorow.

**Comment by dawidmalina** [comment]
Done

**Pull request merged**